SolidCP.Installer/Sources/SolidCP.Setup/Wizard/CertificatePage.cs

```diff
@@ -0,0 +1,250 @@
+
+		protected override void InitializePageInternal()
+		{
+			base.InitializePageInternal();
+
+			Text = "Certificate Settings";
+
+			string component = SetupVariables.ComponentFullName;
+			Description = $"Configure a Server Certificate for {component}.";
+
+			AllowMoveBack = true;
+			AllowMoveNext = true;
+			AllowCancel = true;
+
+			// init fields
+			txtLetsEncryptEmail.Text = SetupVariables.LetsEncryptEmail ?? "";
+			txtCertFileFile.Text = SetupVariables.CertificateFile ?? "";
+			txtCertFilePassword.Text = SetupVariables.CertificatePassword ?? "";
+			txtStoreLocation.Text = SetupVariables.CertificateStoreLocation ?? "";
+			txtStoreName.Text = SetupVariables.CertificateStore ?? "";
+			txtStoreFindType.Text = SetupVariables.CertificateFindType ?? "";
+			txtStoreFindValue.Text = SetupVariables.CertificateFindValue ?? "";
+			manualCert.Checked = false;
+			tabControl.Selected += SetAllowedMoveNext;
+			manualCert.CheckedChanged += SetAllowedMoveNext;
+
+			txtStoreLocation.Items.Clear();
+			txtStoreLocation.Items.Add(StoreLocation.LocalMachine.ToString());
+			txtStoreLocation.Items.Add(StoreLocation.CurrentUser.ToString());
+
+
+			txtStoreName.Items.Clear();
+			txtStoreName.Items.Add(StoreName.Root.ToString());
+			txtStoreName.Items.Add(StoreName.My.ToString());
+			txtStoreName.Items.Add(StoreName.TrustedPeople.ToString());
+
+			txtStoreFindType.Items.Clear();
+			txtStoreFindType.Items.Add(X509FindType.FindBySubjectName.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindByThumbprint.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindBySubjectDistinguishedName.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindBySubjectKeyIdentifier.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindBySerialNumber.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindByIssuerName.ToString());
+			txtStoreFindType.Items.Add(X509FindType.FindByIssuerDistinguishedName.ToString());
+
+			txtCertFileFile.Enabled = txtCertFilePassword.Enabled = btnOpenCertFile.Enabled = !OSInfo.IsWindows;
+
+			if (OSInfo.IsWindows) tabControl.Controls.Remove(tabPageCertFile);
+
+			Update();
+		}
+
+		private void SetAllowedMoveNext(object sender, EventArgs args)
+		{
+			AllowMoveNext = !IsHttps || (tabControl.SelectedTab != tabPageManual) || manualCert.Checked;
+		}
+
+		bool iis7 => SetupVariables.IISVersion.Major >= 7;
+
+		bool IsHttps => (iis7 || !OSInfo.IsWindows) && Utils.IsHttps(SetupVariables.WebSiteIP, SetupVariables.WebSiteDomain);
+
+		public override bool Hidden => !IsHttps;
+		protected internal override void OnAfterDisplay(EventArgs e)
+		{
+			base.OnAfterDisplay(e);
+			//unattended setup
+			if ((!string.IsNullOrEmpty(Wizard.SetupVariables.SetupXml) || !IsHttps) && AllowMoveNext)
+				Wizard.GoNext();
+		}
+
+		private bool CheckEmail()
+		{
+			string email = txtLetsEncryptEmail.Text?.Trim() ?? "";
+
+			if (!Regex.IsMatch(email, "^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,6})$"))
```

Comment:
This is a bad idea to check emails via regex :) 
better to use a special library that can check RFC email standard, or do not check email at all.
even this https://emailregex.com/index.html regex not works in 0.01%

Reply:
Thanx, Changed the code to:
`if (!email.Contains("@"))`